pkg/cache/cluster.go

```diff
@@ -32,34 +31,6 @@ const (
 	ClusterRetryTimeout        = 10 * time.Second
 )
 
-var (
-	// listSemaphore is used to limit the number of concurrent k8s list queries.
-	// Global limit is required to avoid memory spikes during cache initialization.
-	// The default limit of 50 is chosen based on experiments.
-	listSemaphore = semaphore.NewWeighted(50)
```

Comment:
Global limit on concurrent lists queries allows avoiding memory spike during cache initialization. Here is corresponding Argo CD ticket: https://github.com/argoproj/argo-cd/issues/3241 

@ash2k , is it possible to use `RateLimiter` to achieve a similar result?

Reply:
@alexmt Hm, no, rate limiter does not allow to achieve what is needed. I've reworked what I'm suggesting to do, please see the commit comment.

Reply:
sorry for delay. looking at PR today/tomorrow